src/include/cache.i

```diff
@@ -91,6 +91,48 @@ __wt_cache_dirty_inuse(WT_CACHE *cache)
 }
 
 /*
+ * __wt_cache_status --
+ *	Return if the cache usage exceeds the eviction or dirty targets.
+ */
+static inline void
+__wt_cache_status(WT_SESSION_IMPL *session, int *evictp, int *dirtyp)
```

Comment:
Doing it this way should result in a better decision, the state of the system may have changed between when the eviction server chose a block and when we're choosing to discard the page in eviction.

It's clearly more work, though.